hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -746,6 +786,29 @@ private Pair<Option<String>, JavaRDD<WriteStatus>> writeToSink(JavaRDD<HoodieRec
     return Pair.of(scheduledCompactionInstant, writeStatusRDD);
   }
 
+  protected JavaRDD<QuarantineJsonEvent> getErrorEventsForWriteStatus(JavaRDD<WriteStatus> writeStatusRDD) {
+    HoodieWriteConfig config = writeClient.getConfig();
+
+    return writeStatusRDD
+        .filter(WriteStatus::hasErrors)
+        .flatMap(x -> {
+          Schema schema = Schema.parse(config.getSchema());
+          Properties props = config.getPayloadConfig().getProps();
+          return x.getFailedRecords().stream()
+              .map(z -> {
+                HoodieRecordPayload hoodieRecordPayload = (HoodieRecordPayload)z.getData();
+                String recordStr;
+                try {
+                  recordStr = (String) hoodieRecordPayload.getInsertValue(schema,
+                      props).map(value -> value.toString()).get();
+                } catch (IOException e) {
+                  recordStr = null;
```

Comment:
Should it be a reasonable message to distinguish IOException from Hudi?

Reply:
The function is not used anywhere. Removed it.